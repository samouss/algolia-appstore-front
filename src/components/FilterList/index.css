.FilterList {
  width: 100%;
  background-color: #ffffff;
  border: 1px solid #f2f2f2;
  margin-right: 1vh;
  margin-bottom: 1vh;

  &:last-child {
    margin-bottom: 0;
    margin-right: 0;
  }

  @media (--tablet) {
    margin-bottom: 0;
    width: 25%;
  }
}
