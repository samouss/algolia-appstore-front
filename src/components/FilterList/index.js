import React from 'react';
import PropTypes from 'prop-types';
import './index.css';

const FilterList = ({ customClassName, children }) => (
  <ul
    className={customClassName}
    styleName="FilterList"
  >
    {children}
  </ul>
);

FilterList.propTypes = {
  children: PropTypes.node.isRequired,
  customClassName: PropTypes.string,
};

FilterList.defaultProps = {
  customClassName: '',
};

export default FilterList;
