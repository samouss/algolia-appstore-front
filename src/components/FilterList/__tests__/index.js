import React from 'react';
import { shallow } from 'enzyme';
import FilterList from '../index';

describe('<FilterList />', () => {
  const defaultProps = {};

  it('expect to render', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <FilterList
        {...props}
      >
        <li>Content</li>
        <li>Content</li>
      </FilterList>,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render with customClassName', () => {
    const props = {
      ...defaultProps,
      customClassName: 'custom-class-name',
    };

    const component = shallow(
      <FilterList
        {...props}
      >
        <li>Content</li>
        <li>Content</li>
      </FilterList>,
    );

    expect(component).toMatchSnapshot();
  });
});
