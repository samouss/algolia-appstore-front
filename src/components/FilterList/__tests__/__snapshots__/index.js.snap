// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterList /> expect to render 1`] = `
<ul
  className=""
  styleName="FilterList"
>
  <li>
    Content
  </li>
  <li>
    Content
  </li>
</ul>
`;

exports[`<FilterList /> expect to render with customClassName 1`] = `
<ul
  className="custom-class-name"
  styleName="FilterList"
>
  <li>
    Content
  </li>
  <li>
    Content
  </li>
</ul>
`;
